context/app/routes_cells.py

```diff
@@ -19,6 +21,54 @@ def _get_client(app):
     return Client(app.config['XMODALITY_ENDPOINT'] + '/api/')
 
 
+# TODO: In python 3.9, functools.cache would be better.
+gene_symbols = None
+
+
+def _get_gene_symbols(app):
+    client = _get_client(app)
+    global gene_symbols
+    if gene_symbols is None:
+        gene_symbols = [gene["gene_symbol"] for gene in client.select_genes().get_list()]
+    return gene_symbols
+
+
+protein_ids = None
+
+
+def _get_protein_ids(app):
+    client = _get_client(app)
+    global protein_ids
+    if protein_ids is None:
+        protein_ids = [protein["protein_id"] for protein in client.select_proteins().get_list()]
+    return protein_ids
```

Comment:
How often are we meant to retrieve `gene_symbols`/`protein_ids`?

Reply:
Also, is it worth refactoring? It seems like they could share code.

Reply:
> How often are we meant to retrieve `gene_symbols`/`protein_ids`?

Once per deployment, and then they would be cached in memory. 
- Slow enough that I think it needs to be pre-fetched.
- Results don't change fast enough that I'd worry about caching on the timescale of a week.
- Down the road, might make warming the cache part of the deployment process.

> Also, is it worth refactoring? It seems like they could share code.

There are memoization decorators that we could use... but since it's not a built-in right now, decided not to add too much magic... could revisit this in the future.